samples/csharp/end-to-end-apps/eShopDashboardML/README.md

```diff
@@ -33,15 +33,138 @@ Learn how to set it up in Visual Studio plus further explanations on the code:
-> Daqing Chen, Sai Liang Sain, and Kun Guo, Data mining for the online retail industry: A case study of RFM model-based customer segmentation using data mining, Journal of Database Marketing and Customer Strategy Management, Vol. 19, No. 3, pp. 197â€“208, 2012 (Published online before print: 27 August 2012. doi: 10.1057/dbm.2012.17).
 
+ var reader = new TextLoader(env, new TextLoader.Arguments
+ {
+     Column = new[] {
+         new TextLoader.Column("next", DataKind.R4, 0 ),
+         new TextLoader.Column("productId", DataKind.Text, 1 ),
+         new TextLoader.Column("year", DataKind.R4, 2 ),
+         new TextLoader.Column("month", DataKind.R4, 3 ),
+         new TextLoader.Column("units", DataKind.R4, 4 ),
+         new TextLoader.Column("avg", DataKind.R4, 5 ),
+         new TextLoader.Column("count", DataKind.R4, 6 ),
+         new TextLoader.Column("max", DataKind.R4, 7 ),
+         new TextLoader.Column("min", DataKind.R4, 8 ),
+         new TextLoader.Column("prev", DataKind.R4, 9 )
+     },
+     HasHeader = true,
+     Separator = ","
+ });
 
+
+ var pipeline = new ConcatEstimator(env, "NumFeatures", new[] { "year", "month", "units", "avg", "count", "max", "min", "prev" })
+     .Append(new CategoricalEstimator(env, "CatFeatures", "productId"))
+     .Append(new ConcatEstimator(env, "Features", new[] { "NumFeatures", "CatFeatures" }))
+     .Append(new CopyColumnsEstimator(env, "next", "Label"))
+     .Append(new FastTreeTweedieTrainer(env, "Label", "Features"));
+
+```
+
+#### 2. Train model
+
+After building the pipeline, we train the forecast model by fitting or using the training data with the selected algorithm:
+
+```csharp
+ var model = pipeline.Fit(datasource);
+```
+
+#### 3. Evaluate model
+
+We evaluate the accuracy of the model. Evaluate model with a sample products
+
+```csharp
+ // Read the model that has been previously saved by the method SaveModel
+ var env = new LocalEnvironment(seed: 1);  //Seed set to any number so you have a deterministic environment
+ ITransformer model;
+ using (var file = File.OpenRead(outputModelPath))
+ {
+     model = TransformerChain
+         .LoadFrom(env, file);
+ }
+
+ var predictor = model.MakePredictionFunction<ProductData, ProductUnitPrediction>(env);
+ // Build sample data
+ ProductData dataSample = new ProductData()
+ {
+     productId = "263",
+     month = 10,
+     year = 2017,
+     avg = 91,
+     max = 370,
+     min = 1,
+     count = 10,
+     prev = 1675,
+     units = 910
+ };
+
+ //model.Predict() predicts the nextperiod/month forecast to the one provided
+ ProductUnitPrediction prediction = predictor.Predict(dataSample);
+ Console.WriteLine($"Product: {dataSample.productId}, month: {dataSample.month + 1}, year: {dataSample.year} - Real value (units): 551, Forecast Prediction (units): {prediction.Score}");
+```
+
+
+#### 4. Consume the model
+
+Basically, we load the model, then the data file and finally we make a prediction function.
```

Comment:
Here it is important to point out that the "consume the model" code is the code you actually use in the ASP.NET Core Web app. Then, also provide here what specific .cs file has this code within the ASP.NET Core web app.